### PR TITLE
Fixes varediting associative lists

### LINKED ISSUE #3380: Assigning associative list values in VV doesn't work.
1. Spawn a clothing item
2. View it's variables
3. Go to the armor variable
4. Hit edit
3. Choose one of the variables
2. Choose the lowest option, you know the one
1. Watch as nothing happens and a runtime appears in DD.

### LINKED ISSUE #7275: Can't var-edit an associative list item's value.
Trying to var-edit an associative list item only lets you rename it. If you rename it you're never asked what to set it's value to, it gets set to null.

The only way to change the item's value is to delete and re-create it.
(This came up while trying to fix a bugged smart virus dispenser that wouldn't eject a bottle.)

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #3380
Fixes #7275

### CODE REVIEW

**Comment by Iamgoofball** [comment]
this explains so much when I was trying to edit a list locally a few weeks ago

:+1:

File: code/modules/admin/verbs/modifyvariables.dm [hunk 1]
```diff
@@ -142,15 +142,33 @@ var/list/VVckey_edit = list("key", "ckey")
 		if(confirm != "Continue")
 			return
 
-	var/list/names = sortList(L)
+	var/assoc = 0
+	if(L.len > 0)
+		var/a = L[1]
+		if(!isnum(a) && L[a] != null)
```
Comment (TheDZD) [inline code/modules/admin/verbs/modifyvariables.dm]: Just finished porting this over and I had a few problems with this line. "!isnum(a)" allows types and other non-numerical entries to go through. This causes "L[a]" to runtime due to a bad index. Changing "!isnum(a)" to "istext(a)" seemed to fix the issue.

Comment (AnturK) [inline code/modules/admin/verbs/modifyvariables.dm]: Thanks for the information, will update this one. Wish there was inbuilt function to check if something's hashable/whatever it uses for indexing.

**Comment by Boggart** [comment]
:+1:  Can you please add #7275 to the list of fixed issues up top so it gets closed too?

**Pull request merged**